lib/AST/Decl.cpp

```diff
@@ -5599,6 +5599,53 @@ bool VarDecl::isPropertyMemberwiseInitializedWithWrappedType() const {
       this, /*checkDefaultInit=*/false);
 }
 
+static bool isInitFirstParamEscapingClosure(const ConstructorDecl *init) {
+  Type initTy = init->getInterfaceType();
+  if (!initTy->is<AnyFunctionType>())
+    return false;
+
+  Type resultTy = initTy->castTo<AnyFunctionType>()->getResult();
+  if (!resultTy->is<FunctionType>())
+    return false;
+
+  auto funcTy = resultTy->castTo<FunctionType>();
+  if (funcTy->getNumParams() == 0)
+    return false;
+
+  Type firstParamTy = funcTy->getParams()[0].getPlainType();
+  if (!firstParamTy->is<FunctionType>())
+    return false;
+
+  return (!firstParamTy->castTo<FunctionType>()->isNoEscape());
+}
+
+bool VarDecl::isInnermostPropertyWrapperInitUsesEscapingAutoClosure() const {
+  auto customAttrs = getAttachedPropertyWrappers();
+  if (customAttrs.empty())
+    return false;
+
+  unsigned innermostWrapperIndex = customAttrs.size() - 1;
+  auto wrappedValueInit = getAttachedPropertyWrapperTypeInfo(
+      innermostWrapperIndex).wrappedValueInit;
+  if (!wrappedValueInit)
+    return false;
+
+  auto wrappedValueInitParams = wrappedValueInit->getParameters();
+  if (wrappedValueInitParams->size() == 0)
+    return false;
+
+  auto firstParam = wrappedValueInitParams->get(0);
+  auto firstParamName = firstParam->getName();
+
+  ASTContext &ctx = getASTContext();
+  if (firstParamName != ctx.Id_wrappedValue &&
```

Comment:
Shouldn't this be `||` instead of `&&`?

Reply:
Why do you say that? As it stands, it reads: If the firstParamName is neither "wrappedValue" nor "initialValue", return false, which looks correct to me.

Reply:
Oops, I misread the check, sorry. The check looks correct.